python/pyspark/pandas/tests/test_series.py

```diff
@@ -1652,6 +1652,40 @@ def test_pop(self):
         with self.assertRaisesRegex(KeyError, msg):
             psser.pop(("lama", "speed", "x"))
 
+        pser = pd.Series(["a", "b", "c", "a"], dtype="category")
+        psser = ps.from_pandas(pser)
+
+        if LooseVersion(pd.__version__) >= LooseVersion("1.3.0"):
+            self.assert_eq(psser.pop(0), pser.pop(0))
+            self.assert_eq(psser, pser)
+
+            self.assert_eq(psser.pop(3), pser.pop(3))
+            self.assert_eq(psser, pser)
+        else:
+            # Before pandas 1.3.0, `pop` modifies the dtype of categorical series wrongly.
+            self.assert_eq(psser.pop(0), pser.pop(0))
```

Comment:
I guess we can use a literal as an expected value instead of `pser.pop(0)`.

```py
self.assert_eq(psser.pop(0), "a")
```

Reply:
Sounds good!